### PR TITLE
Feature/SOF-7138: new layout for Python Transformation

### PR DESCRIPTION
**Pull request by VsevolodX**
This PR is to adapt existing PT dialog to new layout: collapsable execution cells based on rendered Python code that can be executed and viewed separately. 
Also all necessary clean ups and updates

### CODE REVIEW

File: src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+                        </Grid>
+                        <Grid item xs={12} md={8}>
+                            <TransformationSelector
+                                // eslint-disable-next-line react/destructuring-assignment
+                                transformation={this.state.transformation}
+                                setTransformation={(newTransformation) =>
+                                    this.setState({ transformation: newTransformation })
+                                }
+                                pythonCode={pythonCode}
+                                url={GITHUB_API_URL}
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx]: Let's make the component accept URL as prop and set the default to this

File: src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+    pythonCode: string;
+    pythonOutput: string;
+    executionCells: ExecutionCellState[];
+}
+
+interface PyodideDataMap {
+    [key: string]: string | PyodideDataMap;
+}
+
+// TODO: merge SOF-7138 into dev and update the URL below
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx]: Remove

File: src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+const GITHUB_API_URL =
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx]: Explain why you are using API and maybe put a docs link, explain what this returns.

```json
[{
name: "README.md",
path: "other/python_transformations/README.md",
sha: "841f5bfd45b5da221d02e4fafc6814c3f868295d",
size: 677,
url: "https://api.github.com/repos/Exabyte-io/api-examples/contents/other/python_transformations/README.md?ref=dev",
html_url: "https://github.com/Exabyte-io/api-examples/blob/dev/other/python_transformations/README.md",
git_url: "https://api.github.com/repos/Exabyte-io/api-examples/git/blobs/841f5bfd45b5da221d02e4fafc6814c3f868295d",
download_url: "https://raw.githubusercontent.com/Exabyte-io/api-examples/dev/other/python_transformations/README.md",
type: "file",
_links: {
self: "https://api.github.com/repos/Exabyte-io/api-examples/contents/other/python_transformations/README.md?ref=dev",
git: "https://api.github.com/repos/Exabyte-io/api-examples/git/blobs/841f5bfd45b5da221d02e4fafc6814c3f868295d",
html: "https://github.com/Exabyte-io/api-examples/blob/dev/other/python_transformations/README.md"
}]
```

Call this `FILE_CONFIGS_FETCH_URL`

File: src/components/3d_editor/advanced_geometry/python_transformation/TransformationSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    title: "Custom Transformation (Empty)",
+    content: `"""Custom Transformation"""\n"""BLOCK: Main"""\ninput_materials=materials_in\n`,
+};
+
+function TransformationSelector(props: TransformationSelectorProps) {
+    const { transformation, setTransformation, setPythonCode, url } = props;
+    const [transformations, setTransformations] = useState<Transformation[]>([]);
+    const [isDataFetched, setIsDataFetched] = useState(false);
+
+    const fetchTransformations = async () => {
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/TransformationSelector.tsx]: Should be in code.js

File: src/stylesheets/_generic.scss [hunk 5]
```diff
@@ -70,3 +70,8 @@ $generic-count: 10;
         -webkit-box-shadow: inset 0 0 10px $gray-light;
     }
 }
+
+code {
```
Comment (timurbazhirov) [inline src/stylesheets/_generic.scss]: This is too generic of a directive, let's remove bootstrap style instead

File: tests-legacy/cucumber/features/menu/advanced/python_transformations.feature [hunk 6]
```diff
@@ -0,0 +1,66 @@
+@watch
```
Comment (timurbazhirov) [inline tests-legacy/cucumber/features/menu/advanced/python_transformations.feature]: Remove

File: src/stylesheets/_generic.scss (hunk 5 above)
Comment (VsevolodX) [reply src/stylesheets/_generic.scss]: We can't edit bootstrap, only override like this. 
We can't remove bootstrap easily, because ThreeJSEditor fullscreen capability seems to be dependent on that. The only place where bootstrap is in effect now.

File: src/stylesheets/_generic.scss [hunk 7]
```diff
@@ -70,3 +70,8 @@ $generic-count: 10;
         -webkit-box-shadow: inset 0 0 10px $gray-light;
     }
 }
+
+#python-transformation-dialog code {
+    color: $white;
+    background-color: $exagray_semi-transparent;
+}
```
Comment (timurbazhirov) [inline src/stylesheets/_generic.scss]: Newline missing

Comment (timurbazhirov) [reply src/stylesheets/_generic.scss]: Move to "_designer.scss"

File: src/utils/fetchFromGitHubAPI.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+            name: file.name,
+            download_url: file.download_url,
+        }));
+    } catch (error) {
+        console.error("Error fetching files:", error);
+        throw error;
+    }
+}
+
+export default fetchFiles;
```
Comment (timurbazhirov) [inline src/utils/fetchFromGitHubAPI.js]: This function should be in code.js

File: src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+            result = await pyodide.runPythonAsync(content, {
+                globals: convertedData,
+            });
+            const { pythonOutput } = this.state;
+            this.updateStateAtIndex(executionCells, sectionIndex, {
+                executionStatus: ExecutionStatus.Ready,
+                output: pythonOutput,
+            });
+        } catch (error: any) {
+            this.setState({ executionStatus: ExecutionStatus.Error });
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx:147]: Clean up eslint complaints

File: src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+            }
+        } catch (error: any) {
+            NPMsAlert.error(error.message);
+        }
+    };
+
+    executeAllExecutionCells = async () => {
+        const { executionCells } = this.state;
+        // eslint-disable-next-line no-restricted-syntax
+        for (const [index] of executionCells.entries()) {
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx]: Rewrite to avoid eslint disable

File: tests/cypress/fixtures/create-interface.py [hunk 11]
```diff
@@ -0,0 +1,199 @@
+"""TITLE: Layer on Substrate."""
```
Comment (timurbazhirov) [inline tests/cypress/fixtures/create-interface.py:1]: We should get this from github

Comment (VsevolodX) [reply tests/cypress/fixtures/create-interface.py:1]: To name files in plain english? Can we use whitespaces tho?
Or should we do `Place%20Layer%20on%20substrate.py`?

Comment (timurbazhirov) [reply tests/cypress/fixtures/create-interface.py:1]: I mean, we should get the file itself from api-examples to have one source of truth, not store a copy here

**Review by timurbazhirov — APPROVED** [review]

File: tests/cypress/fixtures/create-interface.py (hunk 11 above)
Comment (VsevolodX) [reply tests/cypress/fixtures/create-interface.py:1]: Ah, the test fixture. Got it.
I decided to have it here for easier debuging and analysis. Here we test the JS-Py data flow and logic, not particular transformation from GitHub or elsewhere
